scripts/update_api.py

```diff
@@ -489,10 +489,11 @@ def mk_java(java_dir, package_name):
     java_native.write('  public static class UIntArrayPtr { public int[] value; }\n')
     java_native.write('  public static native void setInternalErrorHandler(long ctx);\n\n')
 
-    java_native.write('  static {\n')
-    java_native.write('    try { System.loadLibrary("z3java"); }\n')
-    java_native.write('    catch (UnsatisfiedLinkError ex) { System.loadLibrary("libz3java"); }\n')
-    java_native.write('  }\n')
+    if not mk_util.use_java_external_library_loading():
+        java_native.write('  static {\n')
+        java_native.write('    try { System.loadLibrary("z3java"); }\n')
+        java_native.write('    catch (UnsatisfiedLinkError ex) { System.loadLibrary("libz3java"); }\n')
+        java_native.write('  }\n')
```

Comment:
@kfriedberger This is going to cause problems for me when I implement support for building the Java bindings for the CMake build system. Please try to avoid having ``update_api.py`` depending on code in ``mk_util.py``. You should make ``use_java_external_library`` a parameter to ``mk_java()`` and ``generate_files()`` so that when ``generate_files()`` is called from ``mk_util.py`` you can set the option. You will also need to update the ``main()`` function code in ``update_api.py`` so this option can be controlled from the command line.